auth/jwt.go

```
@@ -38,6 +46,25 @@ type JWTConfigCommon struct {
 	ChannelsClaim string `json:"channels_claim"`
 }
 
+// ValidFor returns whether the issuer matches, and one of the audiences matches
+func (j JWTConfigCommon) ValidFor(issuer string, audiences audience) bool {
+	if j.Issuer != issuer {
+		return false
+	}
+	// Validation enforces that ClientID is non-nil for OIDC.
+	if j.ClientID == nil || *j.ClientID == "" {
+		return true
```

Should this be false?  If validation is supposed to enforce that clientID shouldn't be empty, we're in an unexpected situation if we hit this, and wouldn't expect that we should return valid=true.

---

That's only the case for OIDC (I can see how the comment is misleading though, will update) - for local-JWT, an empty (but non-nil) `ClientID` disables `aud` verification ([spec](https://docs.google.com/document/d/1jxz8fnDfPQ-BXxBGOGEgf2NoTRh00A4v30wPt1NA8hM/edit#heading=h.gyjd8mnd5uke)).

---

The updated comment still sounds like j.ClientID==nil should return false, while j.ClientID="" should return true.  I think this is more than a nitpick, as we want to avoid bugs that fail to set j.ClientID and result in audience validation being skipped.

---

Makes sense. I've changed it to return false, i.e. fail closed rather than open.